src/OpenToolkit.Mathematics/Geometry/Box3.cs

```diff
@@ -0,0 +1,234 @@
+//
+// Box3.cs
+//
+// Copyright (C) 2019 OpenTK
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+//
+
+using System;
+using System.Globalization;
+using System.Runtime.InteropServices;
+
+namespace OpenToolkit.Mathematics
+{
+    /// <summary>
+    /// Defines a 3d box.
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential)]
+    public struct Box3 : IEquatable<Box3>
+    {
+        /// <summary>
+        /// The left boundary of the structure.
+        /// </summary>
+        public float Left;
+
+        /// <summary>
+        /// The right boundary of the structure
+        /// </summary>
+        public float Right;
+
+        /// <summary>
+        /// The top boundary of the structure
+        /// </summary>
+        public float Top;
+
+        /// <summary>
+        /// The bottom boundary of the structure
+        /// </summary>
+        public float Bottom;
+
+        /// <summary>
+        /// The back boundary of the structure
+        /// </summary>
+        public float Back;
+
+        /// <summary>
+        /// The front boundary of the structure
+        /// </summary>
+        public float Front;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="Box3"/> struct.
+        /// </summary>
+        /// <param name="left">The position of the left boundary.</param>
+        /// <param name="top">The position of the top boundary.</param>
+        /// <param name="front">The position of the front boundary.</param>
+        /// <param name="right">The position of the right boundary.</param>
+        /// <param name="bottom">The position of the bottom boundary.</param>
+        /// <param name="back">The position of the back boundary.</param>
+        public Box3(float left, float top, float front, float right, float bottom, float back)
+        {
+            Left = left;
+            Top = top;
+            Front = front;
+            Right = right;
+            Bottom = bottom;
+            Back = back;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Box3"/> struct.
+        /// </summary>
+        /// <param name="frontTopLeft">An OpenToolkit.Vector3 describing the front-top-left corner of the Box3.</param>
+        /// <param name="backBottomRight">An OpenToolkit.Vector3 describing the back-bottom-right corner of the Box3.</param>
+        public Box3(Vector3 frontTopLeft, Vector3 backBottomRight)
```

Comment:
min/max are a lot easier to understand in my honest opinion

Reply:
Will get fixed in #920